src/data/move-attrs/random-moveset-move-attr.ts

```diff
@@ -1,67 +1,64 @@
-   * against a random legal target
-   * @param user the {@linkcode Pokemon} using the move
-   * @param target the {@linkcode Pokemon} targeted by the move
-   * @param _move the {@linkcode Move} being used
-   * @returns `true` if a move is successfully invoked
+   * User calls a random moveId selected in {@linkcode getCondition}
+   * @param user Pokemon that used the move and will call a random move
+   * @param target Pokemon that will be targeted by the random move (if single target)
+   * @param move Move being used
+   * @param args Unused
    */
-  override apply(user: Pokemon, target: Pokemon, _move: Move): boolean {
-    const moveset = (!this.enemyMoveset ? user : target).getMoveset();
-    const moves = moveset.filter((m) => !m.getMove().hasFlag(MoveFlags.IGNORE_VIRTUAL));
-    if (moves.length) {
-      const move = moves[user.randSeedInt(moves.length)];
-      const moveIndex = moveset.findIndex((m) => m.moveId === move?.moveId);
-      const moveTargets = getMoveTargets(user, move.moveId);
-      if (!moveTargets.targets.length) {
-        return false;
+  override apply(user: Pokemon, target: Pokemon, _move: Move, overridden: BooleanHolder): boolean {
+    return super.apply(user, target, allMoves[this.moveId], overridden);
+  }
+
+  override getCondition(): MoveConditionFunc {
+    return (user, _target, _move) => {
+      // includeParty will be true for Assist, false for Sleep Talk
+      let allies: Pokemon[];
+      if (this.includeParty) {
+        allies = user.isPlayer()
+          ? globalScene.getPlayerParty().filter((p) => p !== user)
+          : globalScene.getEnemyParty().filter((p) => p !== user);
+      } else {
+        allies = [user];
       }
-      let selectTargets: BattlerIndex[];
-      switch (true) {
-        case moveTargets.multiple || moveTargets.targets.length === 1: {
-          selectTargets = moveTargets.targets;
-          break;
-        }
-        case moveTargets.targets.indexOf(target.getBattlerIndex()) > -1: {
-          selectTargets = [target.getBattlerIndex()];
-          break;
-        }
-        default: {
-          moveTargets.targets.splice(moveTargets.targets.indexOf(user.getAlly().getBattlerIndex()));
-          selectTargets = [moveTargets.targets[user.randSeedInt(moveTargets.targets.length)]];
-          break;
-        }
+
+      const partyMoveset = allies.map((p) => p.moveset).flat();
+      const moves = partyMoveset.filter(
+        (m) => !this.invalidMoves.includes(m.moveId) && !m.getMove().name.endsWith(" (N)"),
+      );
+
+      if (moves.length === 0) {
+        return false;
       }
-      const targets = selectTargets;
-      user.getMoveQueue().push({ move: move.moveId, targets: targets, ignorePP: true });
-      globalScene.unshiftPhase(new MovePhase(user, targets, moveset[moveIndex], true));
-      return true;
-    }
 
-    return false;
+      this.moveId = moves[user.randSeedInt(moves.length)].moveId;
```

Comment:
```suggestion
      moves.sort((a, b) => a.moveId - b.moveId);
      this.moveId = moves[user.randSeedInt(moves.length)].moveId;
```
Sorting the moves before picking a random move prevents the player from RNG manipulating the selected move by swapping moveset orders in the Pokemon summary UI.